src/pipeline_v3/core/index_manager.py

```
@@ -30,16 +30,18 @@
 
     LLAMA_INDEX_AVAILABLE = True
-except ImportError:
-    LLAMA_INDEX_AVAILABLE = False
-
-# Add parent directory to path for imports
-import sys
+except ImportError as e:
+    import logging
 
-sys.path.append(str(Path(__file__).parent.parent))
+    logging.exception(f"Failed to import LlamaIndex dependencies: {e}")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)